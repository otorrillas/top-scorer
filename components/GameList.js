import React from 'react';
import { connect } from 'react-redux';

import _map from 'lodash/map';

import { getGames } from '../store/selectors';

import GameItem from './GameItem';

import styles from './GameList.css';

const GameList = ({ games }) => {
  console.log(games);
  return (
    <ul className={styles.gameList}>
      {_map(games, ({ id, abbreviation: name, assets }) => (
        <li key={id} className={styles.gameListItem}>
          <GameItem id={id} name={name} logo={assets['cover-medium'].uri} />
        </li>
      ))}
    </ul>
  );
};

const mapStateToProps = state => ({
  games: getGames(state)
});

const mapDispatchToProps = dispatch => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(GameList);
