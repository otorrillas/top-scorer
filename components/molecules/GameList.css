.gameListItem {
  list-style-type: none;
}
