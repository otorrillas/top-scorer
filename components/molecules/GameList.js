import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import _map from 'lodash/map';

import { getGames } from 'Store/selectors';

import { GameShape } from 'Utils/propTypes';

import GameItem from '../atoms/GameItem';

import styles from './GameList.css';

export const GameListView = ({ games }) => {
  return (
    <ul className={styles.gameList}>
      {_map(games, ({ id, abbreviation: name, assets }) => (
        <li key={id} className={styles.gameListItem}>
          <GameItem id={id} name={name} logo={assets['cover-medium'].uri} />
        </li>
      ))}
    </ul>
  );
};

GameListView.propTypes = {
  games: PropTypes.objectOf(GameShape)
};

const mapStateToProps = state => ({
  games: getGames(state)
});

export default connect(
  mapStateToProps,
  null
)(GameListView);
