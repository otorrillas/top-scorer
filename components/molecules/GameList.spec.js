import React from 'react';
import renderer from 'react-test-renderer';

import mockGames from 'Utils/mocks';

import { GameListView } from './GameList';

describe('<GameList />', () => {
  const renderComponent = props => renderer.create(<GameListView {...props} />);

  describe('@render', () => {
    it('default', () => {
      const tree = renderComponent({
        games: mockGames,
        id: '1234'
      });
      expect(tree).toMatchSnapshot();
    });
  });
});
