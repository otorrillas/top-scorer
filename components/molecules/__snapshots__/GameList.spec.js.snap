// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameList /> @render default 1`] = `
<ul
  className="gameList"
/>
`;
